src/components/pages/Courses.js

```diff
@@ -0,0 +1,16 @@
+import React, { Fragment, useEffect } from 'react';
+
+const Courses = ({ getCourses, courses }) => {
+  useEffect(() => {
+    getCourses();
+    // eslint-disable-next-line
+  }, []);
```

Comment:
I think the second argument here is optional especially if there is no arguments. Let's remove the empty array as second param here.

Reply:
I'm actually confused with the purpose of second param. What is the purpose of it?

Reply:
function inside `useEffect` is outside the scope of the component itself. But sometimes you need a variable inside the scope of the component to be used in `useEffect` for some changes to trigger. So when you do, you can pass in those variables in that array:

```js
const [offset, setOffset] = useState(0);

useEffect(( /* (Implied example:) [offset] */ ) => { <-- that's how it binds the variable scope
  // on scroll to bottom or pagination click
  Courses.fetch({
    ...,
    offset,
  });
}, [offset]);
```

Otherwise, "offset" inside `useEffect` function cannot find `offset` variable.